var/spack/repos/builtin/packages/paraview/package.py

```diff
@@ -220,6 +219,14 @@ class Paraview(CMakePackage, CudaPackage):
     # https://gitlab.kitware.com/vtk/vtk/-/merge_requests/8653
     patch('vtk-adios2-module-no-kit.patch', when='@5.8:5.10')
 
+    # The ninja generator has problems with XL and CCE
+    # https://gitlab.kitware.com/paraview/paraview/-/issues/21223
+    @property
+    def generator(self):
+        if self.spec.satisfies('%cce') or self.spec.satisfies('%xl') or self.spec.satisfies('%xl_r'):
```

Comment:
There's one issue with this implementation: we have no dependency on `gmake` and we have an unconditional dependency on `ninja`. Adding a conditional dependency on `gmake` would be easy, but removing the one on `ninja` I think requires negation (which is currently not possible).

I wonder if turning `generator` into a sticky variant is possible in principle :thinking:

Reply:
Hey, I have a possible solution. Let me test it with a couple of scenarios.

Reply:
I am not a huge fan of adding more variants, especially since this doesn't actually change the binaries that are built.

It would be better to either understand why these compilers don't work with these compilers and fix that or put some of this logic into the `CMakePackage`.

Reply:
> @quellyn
> 
> > the patch is just falling back to 'Unix Makefiles' for cce and xl builds only
> 
> I get that, I think that is a fine solution. I think we should explore further why are those compilers not working with Ninja. If there is not fix maybe this should go in the `CMakePackage`, then any other packages that need to use Ninja because builds take over a day without it should not have to duplicate/maintain this fix.

Definitely, the preferred solution would be to resolve the underlying issues with Ninja and the XL and CCE compilers. I think that will take more time and expertise than I have, however.

Really, this PR just seeks to restore basic functionality that was lost after the adoption of #28734.